std/range/package.d

```diff
@@ -9594,6 +9594,514 @@ auto refRange(R)(R* range)
     foo(r);
 }
 
+/**
+Bitwise adapter over integral type ranges. Consumes the range elements bit by bit.
+*/
+struct Bitwise(R)
+    if (isInputRange!R && isIntegral!(ElementType!R))
+{
+private:
+    alias ElemType = ElementType!R;
+    alias UnsignedElemType = Unsigned!ElemType;
+
+    R parent;
+
+    enum bitsNum = ElemType.sizeof * 8;
+    size_t maskPos = bitsNum;
+
+    static if (hasLength!R)
+    {
+        ulong len;
```

Comment:
Let's delete this field and compute it on the fly from the source's length and the mask(s). It looks like more work but it's simpler and may actually be faster.